### PR TITLE
Added SubTypesScanner when ReflectionsCache is instantiated

### PR DESCRIPTION
**Pull request by janghwan**
Added SubTypesScanner when ReflectionsCache is instantiated since
getTypesAnnotatedWith method requires SubTypesScanner to be configured since Reflections version 0.9.9.

http://reflections.googlecode.com/svn/trunk/reflections/javadoc/apidocs/org/reflections/Reflections.html#getTypesAnnotatedWith(java.lang.Class)

### CODE REVIEW

**Comment by typesafehub-validator** [comment]
Hi @janghwan,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Typesafe Contributors License Agreement:

[http://www.typesafe.com/contribute/cla](http://www.typesafe.com/contribute/cla)

**Comment by marcospereira** [comment]
@janghwan is it possible to add a test case/spec here?

Thanks!

**Comment by janghwan** [comment]
Hi @marcospereira 
While writing test specs I found that `ReflectionsCache` is used only in test mode, and `play.libs.Classpath.getReflections` has the same duplicated code for run time, so I had to fix those lines too. (I didn't notice it because the plugin I currently work on uses `ReflectionsCache` directly)

There is no test case for `Classpath` either, so it seems it has never been tested. I confirmed `Classpath.getTypesAnnotatedWith` is not working properly currently. Thus, I'm trying to write test case for it now. But the problem is, methods in `Classpath` require `Application` object as a parameter. To test the method, I need to mock `Application` object. Do you have any test example which is using this kind of mock application?

**Comment by janghwan** [comment]
I fixed related issues in `play.libs.Classpath` also, and added test case for the class.

**Comment by janghwan** [comment]
The original issue was reported to mailing list.
https://groups.google.com/forum/#!topic/play-framework/onv5GIYnNis

File: framework/project/Build.scala [hunk 1]
```diff
@@ -321,6 +321,7 @@ object PlayBuild extends Build {
   lazy val PlayJavaProject = PlayCrossBuiltProject("Play-Java", "play-java")
     .settings(libraryDependencies ++= javaDeps ++ javaTestDeps)
     .dependsOn(PlayProject % "compile;test->test")
+    .dependsOn(PlaySpecs2Project % "test")
```
Comment (marcospereira) [inline framework/project/Build.scala]: Do we really need this dependency here?

File: framework/src/play-java/src/main/java/play/libs/Classpath.java [hunk 2]
```diff
@@ -95,7 +95,7 @@ private static Reflections getReflections(Environment env, String packageName) {
                 new ConfigurationBuilder()
                     .addUrls(ClasspathHelper.forPackage(packageName, env.classLoader()))
                     .filterInputsBy(new FilterBuilder().include(FilterBuilder.prefix(packageName + ".")))
-                    .setScanners(new TypeElementsScanner(), new TypeAnnotationsScanner()));
+                    .setScanners(new TypeElementsScanner(), new TypeAnnotationsScanner(), new SubTypesScanner()));
```
Comment (marcospereira) [inline framework/src/play-java/src/main/java/play/libs/Classpath.java]: Can we remove these duplications?

**Comment by marcospereira** [comment]
@janghwan thanks!

Could you please take a look at the comments I've just made? Also, please, squash your commits into a single one.

File: framework/project/Build.scala (hunk 1 above)
Comment (janghwan) [inline framework/project/Build.scala]: I changed it to `PlayTestProject` that is more appropriate.
`ClasspathTest` requires fake app instance as parameter, which rely on `PlayTestProject`

**Comment by janghwan** [comment]
@marcospereira 
I refactored duplicate code as a method, and squashed commits into a single one.
Thanks for help.

**Comment by laran** [comment]
What version of play was/will this be released in? Is there a workaround for Play 2.4? I'm having the same issue.

**Comment by marcospereira** [comment]
@laran Play 2.5.x. This was not backported to 2.4.x branch.

**Pull request merged**